File: articles/bounded-rationality-economic-decision-making/fortran/bounded_rationality_search.f90
program bounded_rationality_search
  implicit none
  real, dimension(6) :: values
  real :: aspiration, search_cost, optimal, chosen_value, net_value
  integer :: i, chosen_index

  values = (/0.20, 0.35, 0.62, 0.75, 0.91, 0.55/)
  aspiration = 0.70
  search_cost = 0.02

  optimal = maxval(values)
  chosen_index = size(values)
  chosen_value = values(size(values))

  do i = 1, size(values)
    if (values(i) >= aspiration) then
      chosen_index = i
      chosen_value = values(i)
      exit
    end if
  end do

  net_value = chosen_value - search_cost * chosen_index

  print *, "Chosen index:", chosen_index
  print *, "Chosen value:", chosen_value
  print *, "Optimal value:", optimal
  print *, "Net value:", net_value
  print *, "Optimization gap:", optimal - chosen_value

end program bounded_rationality_search

File: articles/bounded-rationality-economic-decision-making/rust/bounded_rationality_search.rs
fn main() {
    let values = vec![0.20, 0.35, 0.62, 0.75, 0.91, 0.55];
    let aspiration = 0.70;
    let search_cost = 0.02;

    let optimal = values.iter().cloned().fold(f64::NEG_INFINITY, f64::max);

    let mut chosen_index = values.len();
    let mut chosen_value = *values.last().unwrap();

    for (i, value) in values.iter().enumerate() {
        if *value >= aspiration {
            chosen_index = i + 1;
            chosen_value = *value;
            break;
        }
    }

    let net_value = chosen_value - search_cost * chosen_index as f64;

    println!("Chosen index: {}", chosen_index);
    println!("Chosen value: {:.3}", chosen_value);
    println!("Optimal value: {:.3}", optimal);
    println!("Net value: {:.3}", net_value);
    println!("Optimization gap: {:.3}", optimal - chosen_value);
}
